moveit_ros/moveit_servo/tests/test_utils.cpp

```diff
@@ -0,0 +1,255 @@
+/* Author    : V Mohammed Ibrahim
+   Desc      : Tests for utilities that depend on the robot/ robot state.
+   Title     : test_utils.cpp
+   Project   : moveit_servo
+   Created   : 06/20/2023
+*/
+
+#include <gtest/gtest.h>
+#include <tf2_eigen/tf2_eigen.hpp>
+
+#include "servo_cpp_fixture.hpp"
+#include <moveit_servo/utils.hpp>
+
+namespace
+{
+
+TEST(ServoUtilsUnitTests, validVector)
+{
+  Eigen::VectorXd valid_vector(7);
+  valid_vector << 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0;
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_vector));
+}
+
+TEST(ServoUtilsUnitTests, invalidVector)
+{
+  Eigen::VectorXd invalid_vector(6);
+  invalid_vector << 0.0, 0.0, 0.0, 0.0, std::nan("");
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_vector));
+}
+
+TEST(ServoUtilsUnitTests, validTwist)
+{
+  Eigen::VectorXd valid_vector(6);
+  valid_vector << 0.0, 0.0, 0.0, 0.0, 0.0, 0.0;
+  moveit_servo::Twist valid_twist{ "panda_link0", valid_vector };
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_twist));
+}
+
+TEST(ServoUtilsUnitTests, emptyTwistFrame)
+{
+  Eigen::VectorXd invalid_vector(6);
+  invalid_vector << 0.0, 0.0, 0.0, 0.0, std::nan("");
+  moveit_servo::Twist invalid_twist;
+  invalid_twist.velocities = invalid_vector;
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_twist));
+}
+
+TEST(ServoUtilsUnitTests, invalidTwistVelocities)
+{
+  Eigen::VectorXd invalid_vector(6);
+  invalid_vector << 0.0, 0.0, 0.0, 0.0, 0.0, std::nan("");
+  moveit_servo::Twist invalid_twist{ "panda_link0", invalid_vector };
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_twist));
+}
+
+TEST(ServoUtilsUnitTests, validIsometry)
+{
+  Eigen::Isometry3d valid_isometry;
+  valid_isometry.setIdentity();
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_isometry));
+}
+
+TEST(ServoUtilsUnitTests, invalidIsometry)
+{
+  Eigen::Isometry3d invalid_isometry;
+  invalid_isometry.setIdentity();
+  invalid_isometry.translation().z() = std::nan("");
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_isometry));
+}
+
+TEST(ServoUtilsUnitTests, validPose)
+{
+  Eigen::Isometry3d valid_isometry;
+  valid_isometry.setIdentity();
+  moveit_servo::Pose valid_pose{ "panda_link0", valid_isometry };
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_pose));
+}
+
+TEST_F(ServoCppFixture, testVeclocityScaling)
+{
+  moveit::core::JointBoundsVector joint_bounds = joint_model_group_->getActiveJointModelsBounds();
+  // Get the upper bound for the velocities of each joint.
+  Eigen::VectorXd incoming_velocities(joint_bounds.size());
+  for (size_t i = 0; i < joint_bounds.size(); i++)
+  {
+    const auto joint_bound = (*joint_bounds[i])[0];
+    if (joint_bound.velocity_bounded_)
+    {
+      incoming_velocities(i) = joint_bound.max_velocity_;
+    }
+  }
+
+  // Create incoming velocities with only joint 1 and joint 7 over limit by a factor of 0.1 and 0.05
+  // Scale all other joint velocities by 0.3 to keep it within limits.
+  incoming_velocities(0) += (incoming_velocities(0) * 0.1);
```

Comment:
Can you do e.g. this instead?

```
incoming_velocities(0) *= 1.1;
```

Reply:
Yup, was going to change it but looks like I forgot. Thanks!

Reply:
Done